### ISSUE TITLE
Better documentation about compiling FW

### ISSUE THREAD

**Issue by JoshSK** [device-farm/nanook-mb#1]
Hello,

could you write a few lines about compiling the firmware, please?
I want to use your code with STM32F411 MiniF4  (blackpill) and/or STM32F103 bluepill.
I have a poblem with libs/dependencies...
As a stepper driver I would like to use DRV8825.
Sorry, I'm new to STM32CubeIDE. 
BTW what toolchain do you use?

Thanks so much

**Comment by burgrp** [comment]
Hi, I'm on vacation this week. Will get back to you next week. PB.

**Comment by burgrp** [comment]
@JoshSK , I have added instruction how to compile the firmware: https://github.com/device-farm/nanook-mb#compiling-firmware
Although, we do not have peripheral libraries generated from SVD available for STM32F411 and STM32F103, it should not be problem to generate them. Please check "silicon svd" command.
I'm quite interested what is your intention - are you going to run the firmware on a different board than nanook-mb?

**Comment by JoshSK** [comment]
Thanks so much. I'll look at your silicon library.
Yes, I would modify your design using "mainstream" blocks like *pills, motor drivers for 3d printers... all these parts on the headers and then I would modify topology for use with multiple EEV (1+n). What you thing about this draft?
Thanks

![nanook](https://user-images.githubusercontent.com/24977065/85690001-6bbb1980-b6d3-11ea-821b-8ada7a937a7d.PNG)

**Comment by burgrp** [comment]
Looks great. I'm currently busy by another 10kW heat pump, which will be based on current nanook-mb board, but I was already thinking about making the electronic more modular. My idea is to split the design to several blocks (PSU, Linux MCU, analog interface, power switches...) each on it's own PCB with a common bus interface with 12/5/3.3V, I2C. Boards would have a standard dimension on one side, with board-to-board connectors, so they can be mounted side by side without additional wiring. Perhaps we could cooperate on that, if you like.

**Comment by JoshSK** [comment]
Hello, that sounds good. Do you mean something like this? 
![image](https://user-images.githubusercontent.com/24977065/86256271-69b3f780-bbb8-11ea-9556-1d77d85ab298.png)

**Comment by burgrp** [comment]
Exactly! There would be two Stepper (MotorControl) boards, one per EEV, but that's a detail.

**Comment by JoshSK** [comment]
Hello again,

Sorry for OT, but what you think about switch communication to CAN bus? For example something like this https://www.kickstarter.com/projects/1242572682/nocan-the-wired-iot-platform-for-makers

**Comment by burgrp** [comment]
Hi,

I would like to stick with I2C because of simplicity. CAN bus needs
additional driver chip and an MCU which has that peripheral. Although there
are obvious advantages of CAN over I2C, there is no need for them.
Communication with sensors does not need to be super fast, there are no
long distances, so signal noise is not an issue. Or, do you have a strong
reason to use CAN, please?
